MediaTracker/app/src/main/java/com/izor066/android/mediatracker/ui/SearchGoogleBooks.java

```diff
@@ -0,0 +1,206 @@
+    EditText searchGoogleBooks;
+    private SearchTask task;
+    List<Book> resultsToAdd = new ArrayList<Book>();
+    SearchResultsAdapter searchResultsAdapter;
+    RecyclerView recyclerView;
+    ProgressBar pb;
+
+    SimpleDateFormat simpleDateformat = new SimpleDateFormat("yyyy-MM-dd");
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_search_google_books);
+
+        searchGoogleBooks = (EditText) findViewById(R.id.et_search_googlebooks);
+
+        searchGoogleBooks.setOnEditorActionListener(this);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        Book placeholderBook = MediaTrackerApplication.getSharedDataSource().getAllBooks().get(0);
+        resultsToAdd.add(placeholderBook);
+
+        searchResultsAdapter = new SearchResultsAdapter(resultsToAdd, this);
+
+        recyclerView = (RecyclerView) findViewById(R.id.rv_search_results);
+
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+
+        recyclerView.setAdapter(searchResultsAdapter);
+
+        recyclerView.setVisibility(View.GONE);
+
+        pb = (ProgressBar) findViewById(R.id.pb_search_books_item);
+        pb.setVisibility(View.GONE);
+
+
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                finish();
+                return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    public boolean onCreateOptionsMenu(Menu menu) {
+        return true;
+    }
+
+    @Override
+    public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+        if (actionId == EditorInfo.IME_ACTION_SEARCH) {
+            searchString = searchGoogleBooks.getText().toString();
+            Toast.makeText(this, "Search for: " + searchString, Toast.LENGTH_SHORT).show();
+            task = new SearchTask();
+            task.execute(searchString);
+            InputMethodManager inputMethodManager = (InputMethodManager)  this.getSystemService(Activity.INPUT_METHOD_SERVICE);
+            inputMethodManager.hideSoftInputFromWindow(this.getCurrentFocus().getWindowToken(), 0);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void onResultClick(Book book) {
+        Toast.makeText(this, "View details for: " + book.getTitle(), Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public void onResultAddClick(Book book) {
+        Toast.makeText(this, "Added entry: " + book.getTitle(), Toast.LENGTH_SHORT).show();
+        MediaTrackerApplication.getSharedDataSource().insertBookToDatabase(book);
+    }
+
+    private class SearchTask extends AsyncTask<String, Void, Volumes> {
```

Comment:
Inner classes should be `static`. If they're not they hold a reference to the outer class (in our case the `SearchGoogleBooks`) and potentially leak it (which means holding in memory that instance even if it's not needed anymore which in turn can cause exceptions). Once you make this class `static` you'll notice that all of the objects that you're using won't be accessible as they're not static (`resultsToAdd`, `searchResultsAdapter`, etc), in this case the solution is to pass all of these dependencies in the constructor of `SearchTask`. As an example:

```java
private static class SearchTask extends AsyncTask<String, Void, Volumes> {
    private final List<Book> resultsToAdd;
    private final SearchResultsAdapter searchResultsAdapter;
    private final ProgressBar pb;
    private final RecyclerView recyclerView;

    public SearchText(List<Book> resultsToAdd, SearchResultsAdapter searchResultsAdapter, ProgressBar pb, RecyclerView recyclerView) {
    this.resultsToAdd = resultsToAdd;
    this.searchResultsAdapter = searchResultsAdapter;
    this.pb = pb;
    this.recyclerView = recyclerView;
}

// Rest of the methods unchanged below ..... onPreExecute(), doInBackground(String... params), onPostExecute(Volumes searchListResponse)
}
```

So when you'll use this class in `onEditorAction()`: 

```java
SearchTask task = newSearchTask(resultsToAdd, searchResultsAdapter, pb, recyclerView);
```